src/tasks.ts

```diff
@@ -38,7 +38,12 @@ function optionsToRtdbRef(baseRef: any, options?: CallRtdbOptions): any {
   return newRef;
 }
 
+/**
+ * @param data - Data to be set in firestore
+ * @returns Data to be set in firestore with timestamp
+ */
 function getDataWithTimestamps(data: { [key: string]: any }): object {
+  /* eslint-disable no-underscore-dangle */
   return Object.keys(data).reduce<object>((acc, currKey) => {
```

Comment:
I think that if data is `undefined` then this `Object.keys` will throw a `Cannot convert undefined or null to object`, so we might want to switch to `data && Object.keys(data).reduce`

Reply:
I think the source of issue is that (according to the firebase typings) the .set() method doesn't accept undefined:

<img width="836" alt="Screenshot 2020-05-18 at 21 10 22" src="https://user-images.githubusercontent.com/3258250/82255254-3e4cb480-994c-11ea-9338-ad5a6bb86a55.png">

So throwing an error here (line: 184) is probably advisable, and keeping the parameter of the getDataWithTimestamps function as DocumentData... unless we want to allow undefined and let Firestore throw an error.

Side note: this TS error wasn't visible before as the adminInstance param of callFirestore was set to any (I tightened that up).

I've also updated the typing for getDataWithTimestamps to be DocumentData too.

Reply:
I like the idea of having the throw instead of having undefined go throw and letting Firestore throw.

Nice work on tightening the other types